WordPress/src/test/java/org/wordpress/android/ui/reader/usecases/ReaderGetPostUseCaseTest.kt

```diff
@@ -0,0 +1,111 @@
+        blogId = this@ReaderGetPostUseCaseTest.blogId
+    }
+
+    private val readerDiscoverSourcePost = ReaderPost().apply {
+        postId = 3L
+        blogId = 4L
+    }
+
+    private lateinit var useCase: ReaderGetPostUseCase
+
+    @Before
+    fun setUp() {
+        useCase = ReaderGetPostUseCase(TEST_DISPATCHER, readerPostTableWrapper)
+    }
+
+    @Test
+    fun `given feed, when reader post is retrieved, feed post is returned`() = test {
+        whenever(readerPostTableWrapper.getFeedPost(blogId = blogId, postId = postId, excludeTextColumn = false))
+                .thenReturn(readerPost)
+
+        val result = useCase.get(blogId = blogId, postId = postId, isFeed = true)
+
+        assertThat(result).isEqualTo(Pair<ReaderPost?, Boolean>(readerPost, true))
+    }
+
+    @Test
+    fun `given blog, when reader post is retrieved, blog post is returned`() = test {
+        whenever(readerPostTableWrapper.getBlogPost(blogId = blogId, postId = postId, excludeTextColumn = false))
+                .thenReturn(readerPost)
+
+        val result = useCase.get(blogId = blogId, postId = postId, isFeed = false)
+
+        assertThat(result).isEqualTo(Pair<ReaderPost?, Boolean>(readerPost, false))
+    }
+
+    @Test
+    fun `given editor pick discover post is found, when reader post is retrieved, discover source post returned`() =
+            test {
+                val readerPost = createPost(isDiscoverPost = true, discoverType = DiscoverType.EDITOR_PICK)
+                whenever(
+                        readerPostTableWrapper.getBlogPost(blogId = blogId, postId = postId, excludeTextColumn = false)
+                ).thenReturn(readerPost)
+                whenever(
+                        readerPostTableWrapper.getBlogPost(
+                                blogId = readerDiscoverSourcePost.blogId,
+                                postId = readerDiscoverSourcePost.postId,
+                                excludeTextColumn = false
+                        )
+                ).thenReturn(readerDiscoverSourcePost)
+
+                val result = useCase.get(blogId = blogId, postId = postId, isFeed = false)
+
+                assertThat(result).isEqualTo(Pair<ReaderPost?, Boolean>(readerDiscoverSourcePost, false))
+            }
+
+    @Test
+    fun `given non editor pick reader post is found, when reader post is retrieved, reader post is returned`() =
+            test {
+                val readerPost = createPost(isDiscoverPost = true, discoverType = DiscoverType.SITE_PICK)
+                whenever(
+                        readerPostTableWrapper.getBlogPost(blogId = blogId, postId = postId, excludeTextColumn = false)
+                ).thenReturn(readerPost)
+
+                val result = useCase.get(blogId = blogId, postId = postId, isFeed = false)
+
+                assertThat(result).isEqualTo(Pair<ReaderPost?, Boolean>(readerPost, false))
+            }
+
+    private fun createPost(
```

Comment:
Minor (🔍): It seems that this helper test function doesn't need the `isDiscoverPost` parameter as this is always `true`. As such what if that is rewritten to:
```
    private fun createDiscoverPost(
        discoverType: DiscoverType = DiscoverType.OTHER
    ): ReaderPost {
        val post = spy(readerPost)
        // The ReaderPost contains business logic and accesses static classes. Using spy() allows us to use it in tests.
        whenever(post.isDiscoverPost).thenReturn(true)
        val mockedDiscoverData: ReaderPostDiscoverData = mock()
        whenever(post.discoverData).thenReturn(mockedDiscoverData)
        whenever(mockedDiscoverData.discoverType).thenReturn(discoverType)
        whenever(mockedDiscoverData.postId).thenReturn(readerDiscoverSourcePost.postId)
        whenever(mockedDiscoverData.blogId).thenReturn(readerDiscoverSourcePost.blogId)
        return post
    }
```

Reply:
Removed in f469d49